cli/azd/pkg/containerapps/container_app.go

```diff
@@ -0,0 +1,281 @@
+func (cas *containerAppService) GetIngressConfiguration(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	appName string,
+) (*ContainerAppIngressConfiguration, error) {
+	containerApp, err := cas.getContainerApp(ctx, subscriptionId, resourceGroup, appName)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving container app properties: %w", err)
+	}
+
+	var hostNames []string
+	if containerApp.Properties != nil &&
+		containerApp.Properties.Configuration != nil &&
+		containerApp.Properties.Configuration.Ingress != nil &&
+		containerApp.Properties.Configuration.Ingress.Fqdn != nil {
+		hostNames = []string{*containerApp.Properties.Configuration.Ingress.Fqdn}
+	} else {
+		hostNames = []string{}
+	}
+
+	return &ContainerAppIngressConfiguration{
+		HostNames: hostNames,
+	}, nil
+}
+
+// Adds and activates a new revision to the specified container app
+func (cas *containerAppService) AddRevision(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	appName string,
+	imageName string,
+) error {
+	containerApp, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
+	if err != nil {
+		return fmt.Errorf("getting container app: %w", err)
+	}
+
+	// Get the latest revision name
+	currentRevisionName := *containerApp.Properties.LatestRevisionName
+	revisionsClient, err := cas.createRevisionsClient(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	revisionResponse, err := revisionsClient.GetRevision(ctx, resourceGroupName, appName, currentRevisionName, nil)
+	if err != nil {
+		return fmt.Errorf("getting revision '%s': %w", currentRevisionName, err)
+	}
+
+	// Update the revision with the new image name and suffix
+	revision := revisionResponse.Revision
+	revision.Properties.Template.RevisionSuffix = convert.RefOf(fmt.Sprintf("azd-deploy-%d", cas.clock.Now().Unix()))
+	revision.Properties.Template.Containers[0].Image = convert.RefOf(imageName)
+
+	// Update the container app with the new revision
+	containerApp.Properties.Template = revision.Properties.Template
+	containerApp, err = cas.syncSecrets(ctx, subscriptionId, resourceGroupName, appName, containerApp)
+	if err != nil {
+		return fmt.Errorf("syncing secrets: %w", err)
+	}
+
+	// Update the container app
+	err = cas.updateContainerApp(ctx, subscriptionId, resourceGroupName, appName, containerApp)
+	if err != nil {
+		return fmt.Errorf("updating container app revision: %w", err)
+	}
+
+	// If the container app is in multiple revision mode, update the traffic to point to the new revision
+	if *containerApp.Properties.Configuration.ActiveRevisionsMode == armappcontainers.ActiveRevisionsModeMultiple {
+		newRevisionName := fmt.Sprintf("%s--%s", appName, *revision.Properties.Template.RevisionSuffix)
+		err = cas.setTrafficWeights(ctx, subscriptionId, resourceGroupName, appName, containerApp, newRevisionName)
+		if err != nil {
+			return fmt.Errorf("setting traffic weights: %w", err)
+		}
+	}
```

Comment:
This disables zero downtime deployments for multiple revisions mode.

Reply:
also what kind of output is expected from the deploy command to make sure the next `azd provision` doesn't blow away the values set here?

Reply:
As long as your provisioning templates accept a parameter for the current/expected image and your application state is passed through then the next `azd provision` would essentially be a no-op.

Right now azd does not support advanced configuration for multiple revision mode and is targeting for simpler use cases.  These items are on our backlog and will be addressed in a future release.

Reply:
There is also a longer term state management work item that needs to be addressed for CI based scenarios, #1336